module/VuFind/src/VuFind/ILS/Driver/ComposedDriver.php

```diff
@@ -0,0 +1,1351 @@
+<?php
+
+/**
+ * Composed Driver.
+ *
+ * PHP version 8
+ *
+ * Copyright (C) Hebis Verbundzentrale 2023.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  ILSdrivers
+ * @author   Thomas Wagener <wagener@hebis.uni-frankfurt.de>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:plugins:ils_drivers Wiki
+ */
+
+namespace VuFind\ILS\Driver;
+
+use VuFind\Date\DateException;
+use VuFind\Exception\ILS as ILSException;
+
+use function call_user_func_array;
+use function count;
+use function func_get_args;
+use function in_array;
+
+/**
+ * Composed Driver.
+ *
+ * ILS Driver for VuFind to use multiple drivers for different tasks and
+ * combine their results.
+ *
+ * @category VuFind
+ * @package  ILSdrivers
+ * @author   Thomas Wagener <wagener@hebis.uni-frankfurt.de>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:plugins:ils_drivers Wiki
+ */
+class ComposedDriver extends AbstractMultiDriver
+{
+    /**
+     * Name of the main driver
+     *
+     * @var string
+     */
+    protected $mainDriver;
+
+    /**
+     * Constructor
+     *
+     * @param \VuFind\Config\PluginManager $configLoader Configuration loader
+     * @param PluginManager                $dm           ILS driver manager
+     */
+    public function __construct(
```

Comment:
Since the configLoader and driverManager are dependencies of the base class, I think you could move this constructor there. I realize that MultiBackend has a different signature, but you could still move this constructor into the base class, and then simplify the MultiBackend constructor to call parent::__construct with the appropriate parameters.

Reply:
done